src/controllers/accounts/accounts.ts

```diff
@@ -209,14 +212,9 @@ export class AccountsController extends EventEmitter {
   }
 
   async removeAccountData(address: Account['addr']) {
-    this.accounts = this.accounts.filter((acc) => acc.addr !== address)
-
-    if (this.selectedAccount === address) {
-      await this.#selectAccount(this.accounts[0]?.addr)
-    }
-
+    this.accounts = getUniqueAccountsArray(this.accounts.filter((acc) => acc.addr !== address))
```

Comment:
Upon controller load, `this.accounts` are wrapped in `getUniqueAccountsArray`, therefore are guaranteed to be unique. So probably there is no need to wrap `this.accounts` again here?